WebContentMain.cpp

```diff
@@ -1,15 +1,43 @@
+#include <AK/DeprecatedString.h>
+#include <Ladybird/FontPluginLadybird.h>
+#include <Ladybird/ImageCodecPluginLadybird.h>
 #include <LibCore/ArgsParser.h>
 #include <LibCore/EventLoop.h>
 #include <LibCore/SystemServerTakeover.h>
+#include <LibFileSystem/FileSystem.h>
 #include <LibJS/Bytecode/Interpreter.h>
 #include <LibMain/Main.h>
 #include <LibWeb/Bindings/MainThreadVM.h>
+#include <LibWeb/Loader/FrameLoader.h>
+#include <LibWeb/Platform/EventLoopPluginSerenity.h>
 #include <WebContent/ConnectionFromClient.h>
 
+DeprecatedString s_serenity_resource_root;
+
+static ErrorOr<void> platform_init() {
+    s_serenity_resource_root = TRY([]() -> ErrorOr<DeprecatedString> {
+        auto const* source_dir = getenv("SERENITY_SOURCE_DIR");
+        if (source_dir) {
+            return DeprecatedString::formatted("{}/Base", source_dir);
+        }
+        auto* home = getenv("XDG_CONFIG_HOME") ?: getenv("HOME");
+        VERIFY(home);
+        auto home_lagom = DeprecatedString::formatted("{}/.lagom", home);
```

Comment:
So:
* if this is using `XDG_CONFIG_HOME` (i.e. `~/.config`), then it should be `~/.config/lagom`, not `.lagom`, but it should still be `.lagom` if it's in `HOME`
* should this be `.config` or `.share`? I.e. are these things that the user actively configures, or are these just resources that get installed along with Ladybird?

Reply:
> Assuming that you can still open those with ::open :)

Nope, those are part of the executable, you can't `open()` them like that. So yeah it makes sense to leave them as files for now then.